blockchain/source/contracts/Gatekeeper.sol

```diff
@@ -0,0 +1,106 @@
+pragma solidity ^0.4.23;
+
+
+import "zeppelin-solidity/contracts/token/ERC20/StandardToken.sol";
+import "zeppelin-solidity/contracts/ownership/Ownable.sol";
+
+
+contract Gatekeeper is Ownable {
+
+    using SafeMath for uint256;
+
+    constructor(address _token) public{
+        token = StandardToken(_token);
+        owner = msg.sender;
+    }
+
+    StandardToken token;
+
+    uint256 transactionCount = 0;
+
+    uint256 requiredTransactionAmountForBlock = 512;
+
+    mapping(uint256 => bytes32) roots;
+
+    uint256 rootsCount = 0;
+
+    mapping(bytes32 => mapping(bytes32 => bool)) paidTx;
+
+    event PayInTx(address indexed from, uint256 indexed txNumber, uint256 indexed value);
+
+    event PayoutTx(address indexed from, uint256 indexed txNumber, uint256 indexed value);
+
+    event BlockEmitted(uint256 indexed blockNumber);
+
+    event TransactionAmountForBlockChanged(uint256 indexed amount);
+
+    event RootAdded(uint256 indexed id, bytes32 indexed root);
+
+    function SetTransactionAmountForBlock(uint256 _amount) onlyOwner public {
```

Comment:
Can you explain somewhere(maybe in comments) logic of this contract?

Reply:
API description placed at Gate API Spec at SONM API (v2) section

Reply:
API still change after yesterday's conversation)